code/code/sys/indexdata.cc

```diff
@@ -210,181 +149,118 @@ void generate_obj_index()
   db.query("select * from mob");
 
-  while(db.fetchRow()){
-    if (tmpi) {
-      // push the previous one into the stack
-      mob_index.push_back(*tmpi);
-      delete tmpi;
-    }
-
-    // start a new data member
-    tmpi = new mobIndexData();
-    if (!tmpi) {
-      perror("mobIndexData");
-      exit(0);
-    }
-    
-    tmpi->virt = convertTo<int>(db["vnum"]);
-    
-    // read the sstrings
-    tmpi->name = mud_str_dup(db["name"]);
-    tmpi->short_desc = mud_str_dup(db["short_desc"]);
-    tmpi->long_desc = mud_str_dup(db["long_desc"]);
-    tmpi->description = mud_str_dup(db["description"]);
-    
-    long fac=convertTo<int>(db["faction"]);
-    
-    tmpi->faction = fac;
-    
-    long Class=convertTo<int>(db["class"]);
-    float arm=convertTo<int>(db["ac"]);
-    float hp=convertTo<int>(db["hpbonus"]);
-    float daml=convertTo<int>(db["damage_level"]);
-    
-    long lev = (long)((arm + hp + daml) / 3);
-    
-    tmpi->Class = Class;
-    tmpi->level = lev;
-    
-    long race=convertTo<int>(db["race"]);
-    long wgt=convertTo<int>(db["weight"]);
-    
-    tmpi->race = race;
-    tmpi->weight = wgt;
-    
-    long spec=convertTo<int>(db["spec_proc"]);
-    
-    tmpi->spec = spec;
-        
-    long maxe=convertTo<int>(db["max_exist"]);
-    
-    tmpi->max_exist = maxe;
-    
-    // handle some stat counters
-    if (lev <= 5) {
+  while (db.fetchRow()) {
+    int virt = convertTo<int>(db["vnum"]);
+    sstring name = db["name"];
+    sstring short_desc = db["short_desc"];
+    sstring long_desc = db["long_desc"];
+    sstring description = db["description"];
+    long faction = convertTo<int>(db["faction"]);
+    long Class = convertTo<int>(db["class"]);
+    float arm = convertTo<int>(db["ac"]);
+    float hp = convertTo<int>(db["hpbonus"]);
+    float daml = convertTo<int>(db["damage_level"]);
+    long level = static_cast<long>((arm + hp + daml) / 3);
+    long race = convertTo<int>(db["race"]);
+    float weight = convertTo<float>(db["weight"]);
+    int spec = convertTo<int>(db["spec_proc"]);
+    int max_exist = convertTo<int>(db["max_exist"]);
+
+    if (level <= 5)
       stats.mobs_1_5++;
-    } else if (lev <= 10) {
+    else if (level <= 10)
       stats.mobs_6_10++;
-    } else if (lev <= 15) {
+    else if (level <= 15)
       stats.mobs_11_15++;
-    } else if (lev <= 20) {
+    else if (level <= 20)
       stats.mobs_16_20++;
-    } else if (lev <= 25) {
+    else if (level <= 25)
       stats.mobs_21_25++;
-    } else if (lev <= 30) {
+    else if (level <= 30)
       stats.mobs_26_30++;
-    } else if (lev <= 40) {
+    else if (level <= 40)
       stats.mobs_31_40++;
-    } else if (lev <= 50) {
+    else if (level <= 50)
       stats.mobs_41_50++;
-    } else if (lev <= 60) {
+    else if (level <= 60)
       stats.mobs_51_60++;
-    } else if (lev <= 70) {
+    else if (level <= 70)
       stats.mobs_61_70++;
-    } else if (lev <= 100) {
+    else if (level <= 100)
       stats.mobs_71_100++;
```

Comment:
I'd pregenerate an array of appropriate stats for each level, then look up by level (with bounds checking for the last one).

Reply:
I'm not quite sure how to do what you're suggesting here, and in the interest of just getting this PR done with finally I think I'm going to just leave it as-is, if it's OK with you. It works fine despite being ugly code.